src/System.ServiceModel.Primitives/tests/ServiceModel/DuplexClientBaseTest.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+using Xunit;
+
+public class DuplexClientBaseTest
+{
+    [Fact]
+    public static void DuplexClientBase_Ctor_Initializes_State()
+    {
+        InstanceContext context = new InstanceContext(new WcfDuplexServiceCallback());
+        Binding binding = new NetTcpBinding();
+        EndpointAddress endpoint = new EndpointAddress(Endpoints.Tcp_NoSecurity_Callback_Address);
+        MyDuplexClientBase<IWcfDuplexService> duplexClientBase = new MyDuplexClientBase<IWcfDuplexService>(context, binding, endpoint);
+
+        Assert.Equal<EndpointAddress>(endpoint, duplexClientBase.Endpoint.Address);
+        Assert.Equal<CommunicationState>(CommunicationState.Created, duplexClientBase.State);
+
+        duplexClientBase.Abort();
+    }
```

Comment:
We should probably discuss how "sample worthy" the test code should be.  I think we recommend Close() followed by Abort() if an exception occurs (http://www.codeproject.com/Articles/622989/WCF-and-the-Try-Catch-Abort-Pattern ).  If we agree all scenario tests are potential samples, I think we should agree what that acceptable sample pattern is regarding closing channels, channel factories, ClientBase<T> etc.   Don't alter this PR for this -- I just want consistency across our tests.